mesonbuild/mesonmain.py

```diff
@@ -287,6 +288,15 @@ def run(original_args, mainfile):
         print('You have python %s.' % sys.version)
         print('Please update your environment')
         return 1
+    # https://github.com/mesonbuild/meson/issues/3653
+    if sys.platform.lower() == 'msys':
+        mlog.error('This python3 seems to be msys/python on MSYS2 Windows, which is known to have semantics incompatible with Meson')
```

Comment:
This is kind of vague as to the actual problem.  'path semantics'?

Reply:
Sure, I'll change that.